packages/turbo-workspace-convert/src/utils.ts

```diff
@@ -0,0 +1,239 @@
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): PackageJson {
+  const packageJsonPath = path.join(workspaceRoot, "package.json");
+  if (!fs.existsSync(packageJsonPath)) {
+    throw new Error(`no "package.json" found at ${workspaceRoot}`);
+  }
+  const packageJson = JSON.parse(
+    fs.readFileSync(packageJsonPath, "utf8")
+  ) as PackageJson;
+  return packageJson;
+}
+
+function getWorkspacePackageManager({
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): boolean | string {
+  const { packageManager } = getPackageJson({ workspaceRoot });
+  if (packageManager) {
+    const { groups } = PACKAGE_JSON_REGEX.exec(packageManager) || {};
+    if (groups) {
+      return groups["manager"];
+    }
+  }
+  return false;
+}
+
+function getWorkspaceName({
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): string {
+  const packageJson = getPackageJson({ workspaceRoot });
+  if (packageJson.name) {
+    return packageJson.name as string;
+  }
+  const workspaceDirectory = path.basename(workspaceRoot);
+  return workspaceDirectory;
+}
+
+function getPnpmWorkspaces({
+  workspaceRoot,
+}: {
+  workspaceRoot: string;
+}): Array<string> {
+  const workspaceFile = path.join(workspaceRoot, "pnpm-workspace.yaml");
+  if (fs.existsSync(workspaceFile)) {
+    const workspaceConfig = yaml.load(
+      fs.readFileSync(workspaceFile, "utf8")
+    ) as { packages: Array<string> };
+    return workspaceConfig.packages;
+  }
+
+  return [];
+}
+
+function expandWorkspaces({
+  workspaceRoot,
+  workspaceGlobs,
+}: {
+  workspaceRoot: string;
+  workspaceGlobs?: string[];
+}): Array<Workspace> {
+  if (!workspaceGlobs) {
+    return [];
+  }
+  return workspaceGlobs
+    .flatMap((workspaceGlob) => {
+      const workspacePackageJsonGlob = `${workspaceGlob}/package.json`;
+      return glob.sync(workspacePackageJsonGlob, {
+        onlyFiles: true,
+        absolute: true,
+        cwd: workspaceRoot,
+      });
+    })
+    .map((workspacePackageJson) => {
+      const workspaceRoot = path.dirname(workspacePackageJson);
+      const name = getWorkspaceName({ workspaceRoot });
+      return {
+        name,
+        paths: {
+          root: workspaceRoot,
+          packageJson: workspacePackageJson,
+          nodeModules: path.join(workspaceRoot, "node_modules"),
+        },
+      };
+    });
+}
+
+function updateDependencyList({
+  dependencyList,
+  project,
+  to,
+}: {
+  dependencyList: DependencyList;
+  project: Project;
+  to: PackageManagerDetails;
+}): { dependencyList: DependencyList; updated: Array<string> } {
+  const updated: Array<string> = [];
+  project.workspaceData.workspaces.forEach((workspace) => {
+    const { name } = workspace;
+    if (dependencyList[name]) {
+      dependencyList[name] = to.name === "pnpm" ? `workspace:*` : "*";
+      updated.push(name);
+    }
+  });
+
+  return { dependencyList, updated };
+}
+
+function updateDependencies({
+  project,
+  workspace,
+  to,
+  logger,
+  options,
```

Comment:
`rm` to force this back into being a utility function.

```suggestion
  logger,
  options,
```

Reply:
You're right. this is bigger than just a util function, I broke it out and cleaned it up